### PR TITLE
add methods to get Array and Map fields

### PR DESCRIPTION
**Pull request by suraj-qlogic**

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java [hunk 1]
```diff
@@ -118,11 +118,15 @@
   public static final Timestamp TIMESTAMP;
   public static final GeoPoint GEO_POINT;
   public static final Blob BLOB;
+  public static final String[] SINGLE_FIELD_ARRAY = {"foo"};
```
Comment (pmakani) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java:121]: use this in test classes

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 2]
```diff
@@ -94,6 +97,7 @@
   private final SingleField SINGLE_FIELD_OBJECT = LocalFirestoreHelper.SINGLE_FIELD_OBJECT;
   private final AllSupportedTypes ALL_SUPPORTED_TYPES_OBJECT =
       LocalFirestoreHelper.ALL_SUPPORTED_TYPES_OBJECT;
+  public static final String[] SINGLE_FIELD_ARRAY = {"foo"};
```
Comment (pmakani) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: remove this line of code and use Single_Field array from helper class

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 3]
```diff
@@ -1340,6 +1344,32 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
     assertEquals(3.3, (Double) docSnap.get("sum"), DOUBLE_EPSILON);
   }
 
+  @Test
+  public void getFieldArray() throws ExecutionException, InterruptedException {
```
Comment (pmakani) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: testGetFieldArray

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java [hunk 4]
```diff
@@ -1340,6 +1344,32 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
...
+    assertArrayEquals(SINGLE_FIELD_ARRAY, documentSnapshot.getArray(Array_Value).toArray());
+    try {
+      documentSnapshot.getArray(OBJECT_VALUE);
+    } catch (IllegalArgumentException expectedException) {
+      assertEquals("IllegalArgumentException", expectedException.getClass().getSimpleName());
+    }
+  }
+
+  @Test
+  public void getFieldMap() throws ExecutionException, InterruptedException {
```
Comment (pmakani) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: same here

**Review by pmakani — CHANGES_REQUESTED** [review]

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 3 above)
Comment (athakor) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: method name should be `getFieldArrayTest`

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 4 above)
Comment (athakor) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: same here `getFieldMap()` should be `getFieldMapTest()`

Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 3 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: done

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java (hunk 2 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java]: done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java (hunk 1 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java:121]: done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java [hunk 5]
```diff
@@ -255,6 +258,18 @@ public void deserializeBasicTypes() throws Exception {
     assertEquals(Timestamp.ofTimeSecondsAndNanos(5, 6), snapshot.getReadTime());
 
     assertEquals(get(), getAllCapture.getValue());
+    assertArrayEquals(SINGLE_FIELD_ARRAY, snapshot.getArray(Array_Value).toArray());
+    try {
+      snapshot.getArray(OBJECT_VALUE);
+    } catch (IllegalArgumentException exception) {
+      assertEquals("IllegalArgumentException", exception.getClass().getSimpleName());
```
Comment (rahulKQL) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java]: I think something like `assertTrue(exception.getMessage().contains("value"));` would be better.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java [hunk 6]
```diff
@@ -255,6 +258,18 @@ public void deserializeBasicTypes() throws Exception {
...
+    }
+    assertEquals(SINGLE_FIELD_MAP.size(), snapshot.getMap(OBJECT_VALUE).size());
+    try {
+      snapshot.getMap(Array_Value);
+    } catch (IllegalArgumentException exception) {
+      assertEquals("IllegalArgumentException", exception.getClass().getSimpleName());
```
Comment (rahulKQL) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java]: same here

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java (hunk 6 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java]: done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java (hunk 5 above)
Comment (suraj-qlogic) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/DocumentReferenceTest.java]: done.

**Review by rahulKQL — APPROVED** [review]
LGTM

**Review by athakor — APPROVED** [review]

**Review by pmakani — APPROVED** [review]

**Pull request closed without merging**